cloudinit/distros/freebsd.py

```diff
@@ -612,18 +222,12 @@ def apply_locale(self, locale, out_fn=None):
                 util.logexc(LOG, "Failed to restore %s backup",
                             self.login_conf_fn)
 
-    def _bring_up_interface(self, device_name):
```

Comment:
other parts of the code are still using this call, but will now fallback to Linux

Reply:
This method interferes with the network renderers. It's a legacy of the old network configuration system. IMO it should just go away.

Reply:
I'm not 100% sure this is true. I'll attempt to trace the path that I think will lead to this being called, starting in `main_init`:

https://github.com/canonical/cloud-init/blob/8116493950e7c47af0ce66fc1bb5d799ce5e477a/cloudinit/cmd/main.py#L362

https://github.com/canonical/cloud-init/blob/8116493950e7c47af0ce66fc1bb5d799ce5e477a/cloudinit/stages.py#L710

https://github.com/canonical/cloud-init/blob/8116493950e7c47af0ce66fc1bb5d799ce5e477a/cloudinit/distros/__init__.py#L186

https://github.com/canonical/cloud-init/blob/8116493950e7c47af0ce66fc1bb5d799ce5e477a/cloudinit/distros/__init__.py#L374

This is now calling `cloudinit.distros.Distro._bring_up_interface`, which will call `ifup`:

https://github.com/canonical/cloud-init/blob/8116493950e7c47af0ce66fc1bb5d799ce5e477a/cloudinit/distros/__init__.py#L357-L358

I'm not 100% sure why you wouldn't be seeing an issue based on this. Perhaps the datasource you're testing with is a local one, and so doesn't trigger this codepath?

(I don't think this needs to block this PR landing, because it still enables FreeBSD use cases that currently don't work, but it does seem like we should understand if we have a gap here in the near-future.)

Reply:
given all the stuff that apply_network_config does, we should implement our own

Reply:
In this case, we should land your refactoring of cloudinit/net first. This will bring a lot of clarity on the current situation.